subsys/emul/i2c/emul_atmel_at24.c

```diff
@@ -133,30 +132,44 @@ static struct i2c_emul_api at24_emul_api = {
  * This should be called for each AT24 device that needs to be emulated. It
  * registers it with the I2C emulation controller.
  *
- * @param emul Emulation information
+ * @param emulator Emulation information
  * @param parent Device to emulate (must use AT24 driver)
  * @return 0 indicating success (always)
  */
-static int emul_atmel_at24_init(const struct emul *emul,
-				const struct device *parent)
+static int emul_atmel_at24_init(const struct emul *emulator, const struct device *parent)
 {
-	const struct at24_emul_cfg *cfg = emul->cfg;
-	struct at24_emul_data *data = emul->data;
+	const struct at24_emul_cfg *cfg = emulator->cfg;
+	struct at24_emul_data *data = emulator->data;
 
-	data->emul.api = &at24_emul_api;
+	data->emul.api = &bus_api;
 	data->emul.addr = cfg->addr;
-	data->emul.parent = emul;
+	data->emul.target = emulator;
 	data->i2c = parent;
 	data->cur_reg = 0;
 
 	/* Start with an erased EEPROM, assuming all 0xff */
 	memset(cfg->buf, 0xff, cfg->size);
 
-	int rc = i2c_emul_register(parent, emul->dev_label, &data->emul);
-
-	return rc;
+	return 0;
 }
 
+#define EEPROM_AT24_EMUL_DATA(n)                                                                   \
+	struct i2c_emul emul_##n;                                                                  \
+	const struct device *i2c_##n;                                                              \
+	uint32_t cur_reg_##n;
+
+#define EEPROM_AT24_EMUL_DEFINE(n, api)                                                            \
+	EMUL_DEFINE(emul_eeprom_at24_init, DT_DRV_INST(n), &eeprom_at24_emul_cfg_##n,              \
+		    &eeprom_at24_emul_data_##n, &api)
+
+#define EEPROM_AT24_EMUL_I2C(n)                                                                    \
+	EEPROM_AT24_EMUL_DATA(n)                                                                   \
+	static const struct at_24_emul_cfg at_24_emul_cfg_##n = { .bus_label =                     \
+									  DT_INST_BUS_LABEL(n),    \
+								  .reg = at_24_emul_reg_##n,       \
+								  .addr = DT_INST_REG_ADDR(n) };   \
+	EEPROM_AT24_EMUL_DEFINE(n, &api_i2c)
```

Comment:
These macros aren't used.  And line 171 has an error - it points to an incorrect reference (should be `&bus_api`)

Reply:
Fixed.